src/Crud/PagesRequestFactory.php

```diff
@@ -0,0 +1,31 @@
+<?php
+declare(strict_types=1);
+
+namespace Biig\Melodiia\Crud;
+
+use Symfony\Component\HttpFoundation\Request;
+
+class PagesRequestFactory
+{
+    /**
+     * @param Request $request
+     * @return PagesRequest
+     */
+    public static function build(Request $request)
+    {
+        $page       = $request->query->getInt('page', 1);
+        $maxPerPage = $request->attributes->get(CrudControllerInterface::MAX_PER_PAGE_ATTRIBUTE, 30);
+        if (true === $request->attributes->getBoolean(CrudControllerInterface::ALLOW_USER_DEFINED_MAX_PAGE, false)) {
+            $maxPerPageQueryAttribute = $request->attributes->get(CrudControllerInterface::MAX_PER_PAGE_QUERY_ATTRIBUTE, 'max_per_page');
```

Comment:
This configuration by controller disturbs me a lot. It should really be in global configuration. (I mean, having it by controller is a mistake to me)

This implies to use the configuration object of Melodiia for cruds...

Reply:
Hummm it could be configured in melodiia global configuration. But in some cases, you may want to increase this limit for one endpoint (example: leads).

Reply:
It's not about the size, it's about the parameter name!

Reply:
Under which key in configuration, do you want it?